### PR TITLE
posix: fix lseek to DATA or HOLE on directory

### PR DESCRIPTION
**Pull request by szychows**
Implemented lseek to DATA or HOLE on directory.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/305)

### CODE REVIEW

File: src/libpmemfile-posix/lseek.c [hunk 1]
```diff
@@ -199,6 +149,82 @@ lseek_end_directory(PMEMfilepool *pfp, struct pmemfile_inode *inode,
...
+		 *
+		 * offset >= fsize
+		 * From GNU man page: ENXIO if
+		 * "...ENXIO  whence is SEEK_DATA or SEEK_HOLE, and the file
+		 * offset is beyond the end of the file..."
+		 */
+		return -ENXIO;
+	} else if (vinode_is_dir(vinode)) {
+		/* Nothing to do for now */
+	}
```
Comment (marcinslusarz) [inline src/libpmemfile-posix/lseek.c]: "else return -ENXIO;"
just in case we'll support more file types

File: src/libpmemfile-posix/lseek.c [hunk 2]
```diff
@@ -199,6 +149,82 @@ lseek_end_directory(PMEMfilepool *pfp, struct pmemfile_inode *inode,
...
+		 * so as man page states - ENXIO errno applies.
+		 *
+		 * SEEK_HOLE - directory is constant data series,
+		 * so first hole is pointed to by last dirent.
+		 */
+		pmemfile_off_t end = lseek_end_directory(pfp, vinode->inode, 0);
+		if (whence == PMEMFILE_SEEK_DATA) {
+			if (offset < 0)
+				offset = -EINVAL;
+			else if (offset > end)
```
Comment (marcinslusarz) [inline src/libpmemfile-posix/lseek.c]: offset >= end

lseek_end_directory returns offset *after* the last entry, so offset == end is above "file" size

**Comment by krzycz** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305)*

**Comment by szychows** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/libpmemfile-posix/lseek.c, line 181 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/305#-KsNJciLGfy5On5C4rT9-r1-181:-Ksd4vWcGrfAzi7RwHiU:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/9886a7f6eb04d957dc046973c3c406acd474ff50/src/libpmemfile-posix/lseek.c#L181)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

"else return -ENXIO;"
just in case we'll support more file types
</blockquote></details>

Done.

---

*[src/libpmemfile-posix/lseek.c, line 208 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/305#-KsNJciLGfy5On5C4rT9-r1-208:-Ksd4wJx0MH2JLLcp74n:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/9886a7f6eb04d957dc046973c3c406acd474ff50/src/libpmemfile-posix/lseek.c#L208)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

offset >= end

lseek_end_directory returns offset *after* the last entry, so offset == end is above "file" size
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305)*

**Comment by marcinslusarz** [comment]
*[src/libpmemfile-posix/lseek.c, line 181 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/305#-KsNJciLGfy5On5C4rT9-r1-181:-Ksd4qPoNApnGjSxmDoo:b-gst8pn) ([raw file](https://github.com/pmem/pmemfile/blob/9886a7f6eb04d957dc046973c3c406acd474ff50/src/libpmemfile-posix/lseek.c#L181)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

"else return -ENXIO;"
just in case we'll support more file types
</blockquote></details>

well, please put new line after else...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305)*

**Comment by szychows** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/libpmemfile-posix/lseek.c, line 181 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/305#-KsNJciLGfy5On5C4rT9-r1-181:-Ksd5tLGYzZDtmAnWl5M:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/9886a7f6eb04d957dc046973c3c406acd474ff50/src/libpmemfile-posix/lseek.c#L181)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

well, please put new line after else...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305#-:-Ksd6nfrOsoiT5K_oNrh:bnfp4nl)*

**Comment by sarahjelinek** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/305#-:-KsfBGX2SbcRv0iAefOe:bnfp4nl)*

**Pull request merged**